### PR TITLE
Add "benchmark" test

### PR DESCRIPTION
**Pull request by khieta**
*Issue #, if available:*

*Description of changes:*

* Update `cedar` submodule to the latest `main` branch
* Restructuring/renaming in `src/cedar_test_impl.rs`, and corresponding updates to `fuzz/src/lib.rs` and `src/lean_impl.rs`.
* Add an implementation for the `CedarTestImplementation` trait that uses our Rust library functions.
* Add a new `benchmark.rs` test that runs the Lean & Rust implementation on the corpus tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by khieta** [comment]
Not sure why CI is failing. I'm working on some other things right now, but I should be able to come back to this PR next week. Happy for any preliminary reviews in the meantime.

File: cedar-drt/tests/benchmark.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        println!("\tMedian: {:.1} micros", median(value.clone()));
+        println!("\tp90: {:.1} micros", p90(value.clone()));
+        println!("\tp99: {:.1} micros", p99(value));
+    }
+}
+
+#[test]
+// Currently, running this in conjunction with existing tests will cause an error (#227).
+// In order see the printed output from this test, run `cargo test -- --ignored --nocapture`.
+// #[ignore]
```
Comment (cdisselkoen) [inline cedar-drt/tests/benchmark.rs:168]: you don't need `--ignored` if `#[ignore]` is commented out like this.  we should either un-comment-out the `#[ignore]` or remove the `--ignored`

**Review by cdisselkoen — APPROVED** [review]

File: cedar-drt/tests/benchmark.rs (hunk 1 above)
Comment (khieta) [reply cedar-drt/tests/benchmark.rs:168]: Aha that explains the build failure. I guess I commented it out to double check my claim in #227.

File: cedar-drt/fuzz/src/lib.rs [hunk 2]
```diff
@@ -64,23 +67,33 @@ pub fn run_eval_test(
...
+            panic!("Unexpected error for {request}\nError: {err}");
+        }
+        TestResult::Success(response) => {
+            // The definitional interpreter response should be `true`
+            assert!(
+                response,
+                "Incorrect evaluation result for {request}\nExpression:\n{expr}\nEntities:\n{entities}\nExpected value:\n{:?}\n",
+                expected
+            )
         }
```
[the remark this answers is not in the document]
Comment (khieta) [reply cedar-drt/fuzz/src/lib.rs:95]: Nope, added.

File: cedar-drt/src/cedar_test_impl.rs [hunk 3]
```diff
@@ -14,66 +14,101 @@
...
 #[derive(Debug, Deserialize)]
-pub struct InterfaceValidationResult {
-    #[serde(rename = "validationErrors")]
-    pub validation_errors: Vec<String>,
+pub struct TestResponse {
+    /// Actual response
+    pub response: InterfaceResponse,
+    /// Timing info in microseconds. This field is a `HashMap` to allow timing
+    /// multiple components (or none at all).
+    pub timing_info: HashMap<String, u128>,
```
[the remark this answers is not in the document]
Comment (khieta) [reply cedar-drt/src/cedar_test_impl.rs:62]: Good suggestion, thanks.

**Comment by khieta** [comment]
CI is fixed thanks to @cdisselkoen.

**Review by shaobo-he-aws — APPROVED** [review]

**Pull request merged**